app/src/main/java/com/cp/tasky/auth/shared/data/repository/AuthRepositoryImpl.kt

```diff
@@ -5,20 +5,26 @@ import com.cp.tasky.auth.shared.data.mapper.LoginUserMapper
 import com.cp.tasky.auth.shared.domain.AuthRepository
 import com.cp.tasky.auth.shared.domain.model.LoginResponse
 import com.cp.tasky.auth.shared.domain.model.UserCredential
+import com.cp.tasky.core.util.NetworkResult
```

Comment:
Having a core->util package is not allowed with the chosen feature-layer packaging approach. This will work as long as you have a single module, but in a multi-module environment, you'll completely mix your layers that way, since which layer is intended to talk to util? Sounds like every layer :D 

Every single function you have can be attributed to one specific layer (presentation, domain or data). Even if you have util functions or classes that works. You can of course work with util packages, but they have to be layered as well. Few examples:
```
Date formatting util -> core/presentation/util
Success/error wrapper -> core/domain/util
HttpClientExt -> core/data/util
EncryptedSharedPreferencesUtil -> core/data/util
```